inc/page_settings.php

```diff
@@ -49,6 +49,11 @@ static function single(){
 	 * @return null
 	 */
 	static function get( $key = false, $default = null ) {
+		// Prevent issues with other queries overriding the main queries settings.
+		if ( ! is_main_query() ) {
```

Comment:
Will it ever be a problem that a call to `get` wont return a value? Maybe we should just use the check for `! is_main_query()` to prevent only `list( $type, $id ) = self::get_current_page();` from running, but not from preventing `get` from working. I'm not sure, but it seems like this has the best chance of preventing unforeseen issues.

Reply:
I can't think of a situation where this code will need to be used outside of the main query. I can't see the harm in accounting for it just to be safe.